ccg_custom/models/product.py

```diff
@@ -44,3 +44,14 @@ def get_real_name(self, partner_id=False):
             if customerinfo and customerinfo.product_name:
                 return customerinfo.product_name
         return self.name
+
+    @api.multi
+    def get_partner_code_name(self, partner_id):
+        self.ensure_one()
+        if partner_id:
+            for supinfo in self.customer_ids.filtered(lambda x: x.name.id ==
+                                                      partner_id):
+                return {'code': supinfo.product_code or self.default_code,
+                        'name': supinfo.product_name or self.name}
+        res = {'code': self.default_code, 'name': self.name}
+        return res
```

Comment:
por defecto creo que el name es requerido, si no poner x.name and x.name.id == partner_id